packages/cli/src/runIOS/runIOS.js

```diff
@@ -4,79 +4,68 @@
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
  *
+ * @flow
  * @format
  */
 
-/* eslint-disable */
+import type { ContextT } from '../core/types.flow';
 
+// eslint-disable-next-line
 const child_process = require('child_process');
 const fs = require('fs');
 const path = require('path');
 const findXcodeProject = require('./findXcodeProject');
-const findReactNativeScripts = require('../util/findReactNativeScripts');
 const parseIOSDevicesList = require('./parseIOSDevicesList');
 const findMatchingSimulator = require('./findMatchingSimulator');
+const { ProcessError } = require('../util/logger');
 
-const getBuildPath = function(configuration = 'Debug', appName, isDevice, scheme) {
-  let device;
-
-  if (isDevice) {
-    device = 'iphoneos';
-  } else if (appName.toLowerCase().includes('tvos')) {
-    device = 'appletvsimulator';
-  } else {
-    device = 'iphonesimulator';
-  }
-
-  return `build/${scheme}/Build/Products/${configuration}-${device}/${appName}.app`;
-};
-const xcprettyAvailable = function() {
-  try {
-    child_process.execSync('xcpretty --version', {
-      stdio: [0, 'pipe', 'ignore'],
-    });
-  } catch (error) {
-    return false;
-  }
-  return true;
+type FlagsT = {
+  simulator: string,
+  configuration: string,
+  scheme: ?string,
+  projectPath: string,
+  device: ?string,
+  udid: ?string,
+  packager: boolean,
+  verbose: boolean,
+  port: number,
```

Comment:
This type seems wrong all the way. These are arguments, so they're all optional, right? Shouldn't it be typed like this?
```jsx
type FlagsT = {|
  simulator?: string,
  configuration?: string,
  scheme?: string,
  projectPath?: string,
  device?: string,
  udid?: string,
  packager?: boolean,
  verbose?: boolean,
  port?: number,
|};
```

Reply:
It seems, but if you take into consideration default values set by the CLI (see the object at the end where options are defined), it looks good. In this context, e.g. `verbose` is always false, unless set. Same for `port`. We do the same in other commands.

We can add a comment if that's confusing.

Reply:
I mean, if an option is present in the object, why would it be falsy? Maybe that's something coming from commander, I didn't check locally.

Reply:
Just checked, this is what commander returns, so feel free to discard my comment